src/main/java/com/yahoo/validatar/execution/pig/Sty.java

```diff
@@ -0,0 +1,229 @@
+            Result result = query.createResults();
+            // dumpSchema will also, unfortunately, print the schema to stdout.
+            List<FieldDetail> metadata = getFieldDetails(server.dumpSchema(alias));
+            populateColumns(metadata, result);
+            while (queryResults.hasNext()) {
+                populateRow(queryResults.next(), metadata, result);
+            }
+            server.shutdown();
+        } catch (IOException ioe) {
+            log.error("Problem with Pig query: \n" + queryValue, ioe);
+            query.setFailure(ioe.toString());
+        } catch (Exception e) {
+            log.error("Error occurred while processing Pig query: " + queryValue, e);
+            query.setFailure(e.toString());
+        }
+    }
+
+    private void populateColumns(List<FieldDetail> metadata, Result result) throws IOException {
+        if (metadata.isEmpty()) {
+            throw new IOException("No metadata of columns found for Pig query");
+        }
+        metadata.forEach(m -> result.addColumn(m.alias));
+    }
+
+    private void populateRow(Tuple row, List<FieldDetail> metadata, Result result) throws ExecException {
+        if (row == null) {
+            log.info("Skipping null row in results...");
+            return;
+        }
+        for (int i = 0; i < metadata.size(); ++i) {
+            FieldDetail column = metadata.get(i);
+            TypedObject value = getTypedObject(row.get(i), column);
+            log.info("Column: " + column.alias + "\tType: " + column.type +
+                     "\tValue: " + (value == null ? "null" : value.data));
+            result.addColumnRow(column.alias, value);
+        }
+    }
+
+    private TypedObject getTypedObject(Object data, FieldDetail detail) throws ExecException {
+        if (data == null) {
+            return null;
+        }
+        byte type = detail.type;
+        switch (type) {
+            case DataType.BOOLEAN:
+                return TypeSystem.asTypedObject(DataType.toBoolean(data, type));
+            case DataType.INTEGER:
+            case DataType.LONG:
+                return TypeSystem.asTypedObject(DataType.toLong(data, type));
+            case DataType.FLOAT:
+            case DataType.DOUBLE:
+                return TypeSystem.asTypedObject(DataType.toDouble(data, type));
+            case DataType.DATETIME:
+                return TypeSystem.asTypedObject(new Timestamp(DataType.toDateTime(data, type).getMillis()));
+            case DataType.BYTE:
+            case DataType.BYTEARRAY:
+            case DataType.CHARARRAY:
+                return TypeSystem.asTypedObject(DataType.toString(data, type));
+            case DataType.BIGINTEGER:
+            case DataType.BIGDECIMAL:
+                return TypeSystem.asTypedObject(DataType.toBigDecimal(data, type));
+            default:
+                //TUPLE, BAG, MAP, INTERNALMAP, GENERIC_WRITABLECOMPARABLE, ERROR, UNKNOWN, NULL and anything else
```

Comment:
That's right - unless we want to support extracting fields from them or allow operators to work on complex fields. I think we should push all of that extraction logic to the engine itself and require the user to pull out atomic data.